### PR TITLE
[Converter] support flex tensor

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add pad template to handle flex tensors.
tensor-converter supports changing capability from flex to static, or static to flex if downstream element has flex-tensor caps.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_converter/tensor_converter.c [hunk 1]
```diff
@@ -319,6 +319,10 @@ gst_tensor_converter_class_init (GstTensorConverterClass * klass)
     g_strfreev (str_array);
   }
 
+  /* append caps string for flexible tensors */
+  gst_caps_append (pad_caps,
+      gst_caps_from_string (GST_TENSORS_FLEX_CAP_DEFAULT));
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_converter/tensor_converter.c:324]: Can you do this with the style of LINE 302-206 if this capability is built-in?

**Review by myungjoo — COMMENTED** [review]
LGTM except for the (mentioned) possible style inconsistency.

Anyway, questions:
Q1. what happens if the pipeline is ```... static-tensor ! tensor-converter ! static-tensor ...```?
Q2. what happens if the pipeline is ```... flex-tensor ! tensor-converter ! flex-tensor ...```?
Q3. what happens if the pipeline is ```... flex-tensor ! tensor-converter ! static-tensor ...``` and the buffer via flex-tensor at nth frame is 100 bytes while stat-tensor has 10 bytes in its size?
Q4. what happens if the pipeline is ```... flex-tensor ! tensor-converter ! static-tensor ...``` and the buffer via flex-tensor at nth frame is 100 bytes while static-tensor has 1000 bytes in its size?

Corresponding behaviors (Q1~Q4) need to be documented later. (we have tensor type document)

File: gst/nnstreamer/tensor_converter/tensor_converter.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_converter/tensor_converter.c]: updated

**Comment by jaeyun-jung** [comment]
> LGTM except for the (mentioned) possible style inconsistency.
> 
> Anyway, questions:
> Q1. what happens if the pipeline is `... static-tensor ! tensor-converter ! static-tensor ...`?
> Q2. what happens if the pipeline is `... flex-tensor ! tensor-converter ! flex-tensor ...`?
> Q3. what happens if the pipeline is `... flex-tensor ! tensor-converter ! static-tensor ...` and the buffer via flex-tensor at nth frame is 100 bytes while stat-tensor has 10 bytes in its size?
> Q4. what happens if the pipeline is `... flex-tensor ! tensor-converter ! static-tensor ...` and the buffer via flex-tensor at nth frame is 100 bytes while static-tensor has 1000 bytes in its size?
> 
> Corresponding behaviors (Q1~Q4) need to be documented later. (we have tensor type document)

Q1. negotiation failure (converter does not have static caps on sink pad)
Q2. Possible, converter just checkes the incoming buffer size.
Q3 and Q4. Failed, incoming buffer size should be exactly matched with converter property.

I will update wiki and readme about flex tensor later.

**Comment by myungjoo** [comment]
> converter just checkes the incoming buffer

This will probably be future work. However,.... shouldn't Q4 be supported if the user provides (new) properties that declare which dimension to be "expanded"? (we may have a working default value for this as well)

File: gst/nnstreamer/tensor_converter/tensor_converter.c [hunk 2]
```diff
@@ -319,6 +325,9 @@ gst_tensor_converter_class_init (GstTensorConverterClass * klass)
     g_strfreev (str_array);
   }
 
+  /* append caps string for flexible tensors */
+  append_flex_tensor_caps_template (pad_caps);
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_converter/tensor_converter.c:329]: Move to Line 313?

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_converter/tensor_converter.c:329]: updated

**Comment by jaeyun-jung** [comment]
> > converter just checkes the incoming buffer
> 
> This will probably be future work. However,.... shouldn't Q4 be supported if the user provides (new) properties that declare which dimension to be "expanded"? (we may have a working default value for this as well)

add todo to change mem block if incoming data is tensor format.

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**